salt/minion_auth/README.md

```diff
@@ -0,0 +1,47 @@
+# Foreman Salt Minion Authentication
+
+Currently, there are two possibilites to authenticate a newly deployed Minion automatically:
+1. Use the /etc/salt/autosign.conf file which stores the hostnames of acceptable hosts.
+2. Use Salt Autosign Grains for a more secure way which relies on a shared secret key.
+
+This README, handles the second option and how to configure it
+
+## Setup
+Add the content of 'master.snippet' to '/etc/salt/master' which configures the Grains key file on the Master and a Reactor. The Grains file holds the acceptable keys and will be written by the Smart Proxy when a new Minion is deployed. The Reactor initiates an interaction with Foreman Salt if a new Minion was authenticated successfully.
+In case there is already a Reactor configured, you need to adapt it using the options mentioned in 'master.snippet'. The directories given in 'master.snippet' are the default ones. In case you want your files in a different place, you have to change the paths accordingly.
+
+If '/srv/salt' is configured as 'file_roots' in your '/etc/salt/master' config, setup the necessary Salt Runners:
+
+```
+/srv/salt/_runners/foreman_file.py
+/srv/salt/_runners/foreman_https.py
+```
+
+Check if the Reactor ('foreman_minion_auth.sls') is at the appropriated location:
+
+```
+/var/lib/foreman-proxy/salt/reactors/foreman_minion_auth.sls
+```
+
+After changing the Salt Master run:
+
+```
+systemctl restart salt-master
+```
+
+After checking the Reactor and Runners, run the following:
+
+```
+salt-run saltutil.sync_all
+```
+
+## Procedure
+
+1. A new host, configured as Salt Minion, is deployed with Foreman Salt.
+2. Foreman Salt generates a unique key for that Minion and distributes it via the Provisioning Template to the host and via an API call to the Smart Proxy.
+3. The Smart Proxy makes the key available for the Salt Autosign Grains procedure by adding it to the previously defined file (by default: /var/lib/foreman-proxy/salt/grains/autosign_key).
+4. The Salt Master accepts the Minion depending on the key and the corresponding auth reactor is triggered on the Salt Master.
```

Comment:
add one additional step like:
Salt minion will be started and uses configured salt autosign grain to authenticate on salt master.

Reply:
Implemented as suggested.